src/gretel_trainer/benchmark/gretel/models.py

```diff
@@ -1,15 +1,34 @@
 from pathlib import Path
-from typing import Dict, Union
+from typing import Dict, Optional, Tuple, Union
 
+import gretel_trainer
+
+from gretel_client.projects.exceptions import ModelConfigError
 from gretel_client.projects.models import read_model_config
+from gretel_trainer import models
+from gretel_trainer.benchmark.core import BenchmarkException, DataSource, Datatype
 
 GretelModelConfig = Union[str, Path, Dict]
 
 
-def _config(blueprint: str, class_name: str) -> Dict:
-    config = read_model_config(blueprint)
-    config["name"] = f"benchmark-{class_name}"
-    return config
+TRAINER_MODEL_TYPE_CONSTRUCTORS = {
+    "actgan": models.GretelACTGAN,
+    "amplify": models.GretelAmplify,
+    "ctgan": models.GretelACTGAN,
+    "synthetics": models.GretelLSTM,
+
+    # Benchmark GretelAuto sends None model_type to Trainer to trigger dynamic model selection
+    "AUTO": lambda _: None,
+}
+
+
+def _is_compatible(model_key: Optional[str], source: DataSource) -> bool:
```

Comment:
With the current usage of this, it looks like `model_key` will be a `str`, since for auto mode it returns `AUTO`.